diff --git a/tests/MultifieldAdministrationTestCase.test b/tests/MultifieldAdministrationTestCase.test
index 99b1353..864d52c 100644
--- a/tests/MultifieldAdministrationTestCase.test
+++ b/tests/MultifieldAdministrationTestCase.test
@@ -13,7 +13,7 @@ class MultifieldAdministrationTestCase extends DrupalWebTestCase {
 
   public function setUp() {
     parent::setUp(array('multifield', 'field_ui', 'node', 'text'));
-    $account = $this->drupalCreateUser(array('administer content types', 'administer multifield'));
+    $account = $this->drupalCreateUser(array('administer content types', 'administer multifield', 'administer fields'));
     $this->drupalLogin($account);
   }
 
